daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -374,7 +397,8 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
 
           res
         }
-        case VariableInProcess => throw new VariableCircularDefinition(varQName, vrd)
+        case VariableBeingDefined => throw new VariableCircularDefinition(varQName, vrd)
+        case VariableInProcess => throw new VariableSuspended(varQName, vrd)
```

Comment:
That is correct. VariableBeingDefined is used when a defineVariable statement references another variable in its default value. Its state gets set to VBD while we are resolving the variable reference and if we hit another variable that is set to VBD it means there is a circular definition.

VariableInProcess is for variables that are suspended but should complete once the suspension is done.

Reply:
Ah, ok, this makes sense. One thing that is maybe unclear to me then. Above we have this:
```scala
  case VariableDefined if (!variable.value.isDefined && variable.defaultValueExpr.isDefined) => {
    Assert.invariant(maybeState.isDefined)
    variable.setState(VariableBeingDefined)
```
So it seems like a variable goes from VariableDefined to VariableBeingDefined when it's read but has no value (and has a defaultValue). My intuition would say that these pre-runtime variables should have started in the state of VariableBeingDefined if they don't have a value and then switch to VariableDefined when they actually do get a value. Since this doesn't appear to be the case, it would probably be really helpful to add a comment explaining why this state change happens.